sql/api/src/main/scala/org/apache/spark/sql/streaming/StatefulProcessorHandle.scala

```diff
@@ -72,6 +72,26 @@ private[sql] trait StatefulProcessorHandle extends Serializable {
    */
   def getListState[T](stateName: String, valEncoder: Encoder[T]): ListState[T]
 
+  /**
+   * Function to create new or return existing list state variable of given type
+   * with ttl. State values will not be returned past ttlDuration, and will be eventually removed
+   * from the state store. Any state update resets the ttl to current processing time plus
```

Comment:
[nit] Might be better to say -

`Any values in listState which have expired after ttlDuration will not returned on get() and will be eventually removed from the state.`